app/api/routes.py

```diff
@@ -323,3 +326,27 @@ def create_new_apikey(email):
     except Exception as e:
         logger.exception(e)
         return standardize_response(status_code=500)
+
+
+def validate_resource(json):
+    validation_errors = {'errors': {'type': 'validation'}}
+    if not json:
+        message = "A JSON body is required to use this endpoint, but none was given"
+        validation_errors['errors']['message'] = message
+        return validation_errors
+
+    validation_errors['errors']['missing_params'] = []
+
+    required = []
+    for column in Resource.__table__.columns:
```

Comment:
What's an example of this?

Reply:
Basically, the `for` loop goes through each column for the `resources` table and checks for all the ones that are `nullable=False`. There are 4 (but one of them is the `id`, so I explicitly don't check for that). The remaining 3 are "name", "category", and "url". You can't have a resource that doesn't have all 3 of those attributes.

Does that answer your question?

Reply:
After #133 gets merged, there will be 4 required fields: `name`, `category`, `url`, and now `paid`.